cloud/rackspace/rax_images.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# This is a DOCUMENTATION stub specific to this module, it extends
+# a documentation fragment located in ansible.utils.module_docs_fragments
+DOCUMENTATION = '''
+---
+module: rax_images
+short_description: Manipulate Rackspace Cloud Server Images
+description:
+  - Manipulate Rackspace Cloud Server Images
+version_added: "2.0"
+options:
+  name:
+    description:
+      - Name of server from which image will be created
+    default: null
+  id:
+    description:
+      - Server ID from which image will be created
+    default: null
+  image_name:
+    description:
+      - Name of image which will be created
+    required: true
+  wait:
+    description:
+      - wait for the image to be in state 'active' before returning
+    default: "no"
+    choices:
+      - "yes"
+      - "no"
+  wait_timeout:
+    description:
+      - how long before wait gives up, in seconds
+    default: 300
+
+author: Vladimir Saienko
+extends_documentation_fragment: rackspace
+'''
+
+EXAMPLES = '''
+- name: Build server image in Rackspace
+  local_action:
+    module: rax_images
+    credentials: ~/.raxpub
+    name: "{{ inventory_hostname }}"
+    image_name: "{{ image_name }}"
+    region: DFW
+    wait: yes
+    wait_timeout: 300
+'''
+
+try:
+    import pyrax
+    HAS_PYRAX = True
+except ImportError:
+    HAS_PYRAX = False
+
+
+def rax_images(module, name, server_id, image_name, wait=True, wait_timeout=300):
+    changed = False
+
+    cs = pyrax.cloudservers
+    imgs = pyrax.images
+
+    if cs is None:
+        module.fail_json(msg='Failed to instantiate client. This '
+                             'typically indicates an invalid region or an '
+                             'incorrectly capitalized region name.')
+
+    search_opts = {}
+    if name:
+        search_opts = dict(name='^%s$' % name)
+        try:
+            servers = cs.servers.list(search_opts=search_opts)
+        except Exception, e:
+            module.fail_json(msg='%s' % e.message)
+
+    elif server_id:
+        servers = []
+        try:
+            servers.append(cs.servers.get(server_id))
+        except Exception, e:
+            pass
```

Comment:
With collapsing `name` and `id` into server, use `rax_find_server` like it is used at https://github.com/ansible/ansible-modules-core/blob/devel/cloud/rackspace/rax_cbs_attachments.py#L112